### DISCUSSION TITLE
Is there a way to add Provider to root node with new routing system ?

### DISCUSSION THREAD

**Discussion by Dhananjay-JSR** [expo/router#46] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/47073516/193420427-867a5e34-791c-4c18-9b9c-0935dd588060.png)

is it possible to edit entry module and add something like this ?

![image](https://user-images.githubusercontent.com/47073516/193420556-f779674b-74f1-4ab9-bb60-3b298a37db21.png)

**Comment by bndkt** [comment, marked as answer]
You would use a layout route for this, see https://expo.github.io/router/docs/features/layout-routes.

Your (layout).tsx could then look like this:

```ts
import { Children } from "expo-router";
import { Provider } from "../components/Provider";

export default function Layout() {
  return (
    <Provider>
      <Children />
    </Provider>
  );
}
```

**Comment by sturmenta** [comment]
at the present date the names have changed, this is a example using [react-native-toast-message](https://github.com/calintamas/react-native-toast-message/tree/main)

```ts
// app/_layout.tsx

import { Slot } from 'expo-router';
import Toast from 'react-native-toast-message';

export default function RootLayout() {
  return (
    <>
      <Slot />
      <Toast />
    </>
  );
}
```
